src/js/tracks/text-track-display.js

```diff
@@ -277,34 +293,18 @@ class TextTrackDisplay extends Component {
   }
 
   /**
-   * Add an {@link TextTrack} to to the {@link Tech}s {@link TextTrackList}.
+   * Style {@Link TextTrack} activeCues according to {@Link TextTrackSettings}.
    *
    * @param {TextTrack} track
-   *        Text track object to be added to the list.
+   *        Text track object containing active cues to style.
    */
-  updateForTrack(track) {
-    if (typeof window.WebVTT !== 'function' || !track.activeCues) {
-      return;
-    }
-
-    const cues = [];
-
-    for (let i = 0; i < track.activeCues.length; i++) {
-      cues.push(track.activeCues[i]);
-    }
-
-    window.WebVTT.processCues(window, cues, this.el_);
-
-    if (!this.player_.textTrackSettings) {
-      return;
-    }
-
+  updateDisplayState(track) {
     const overrides = this.player_.textTrackSettings.getValues();
 
-    let i = cues.length;
+    let i = track.activeCues.length;
```

Comment:
we should get a reference to the activeCues instead of calling referencing activeCues directly each time

```js
const cues = track.activeCues;
```

then this line and the line below can be unchanged.